docs/compilation/compression.md

```diff
@@ -0,0 +1,43 @@
+**# Compression**
+
+Fully Homomorphic Encryption (FHE) needs both ciphertexts (encrypted data) and evaluation keys to carry out the homomorphic evaluation of a function. Both elements are quite large in size. Depending on the use-case, application deployment, and the method for transmitting and storing ciphertexts and evaluation keys, these sizes could critically affect the application's performance.
+
+During compilation, you can enable compression options to enforce the use of compression features. The two available compression options are:
+
+* **compress\_evaluation\_keys**: bool = False,
+    - This specifies that the compressed form of evaluation keys should be used for serialization.
+* **compress\_input\_ciphertexts**: bool = False,
+    * This specifies that the compressed form of input ciphertexts should be used for serialization.
+
+You can see the impact of compression by comparing the size of the serialized form of input ciphertexts and evaluation keys with a sample code.
+
+```python
+from concrete import fhe
+def test_compression(compression):
+    @fhe.compiler({"counter": "encrypted"})
+    def f(counter):
+       return counter // 2
+
+    circuit = f.compile(fhe.inputset(fhe.tensor[fhe.uint2, 3]),
+                        compress_evaluation_keys=compression,
+                        compress_input_ciphertexts=compression)
+
+    print(f"Sizes with compression = {compression}")
+    print(f" - of the input ciphertext {len(circuit.encrypt(list([0 for i in range(3)])).serialize())}")
+    print(f" - of the evaluation keys {len(circuit.keys.serialize())}")
+
+test_compression(False)
+test_compression(True)
+```
+
+The compression factor largely depends on the cryptographic parameters identified during the compilation and the compression algorithms used.
+
+Currently, it is the seeded compression algorithms that are employed. These algorithms rely on the fact that CSPRNGs are deterministic. Consequently, the chain of random values can be replaced by the seed and later recalculated using the same seed.
+
+Typically, the size of a ciphertext is `(lwe dimension + 1) * 8` bytes, while the size of a seeded ciphertext is constant, equal to `3 * 8` bytes. Thus, the compression factor ranges from a hundred to thousands. Understanding the compression factor of keys is more complex and typically ranges between 0 and 10.
+
+{% hint style="warning" %}
+
+However, everything comes at a cost. While compression may save bandwidth and disk space, it also incurs the cost of decompression. Currently, compression and decompression occur more or less lazily during FHE evaluation without any control.
```

Comment:
First sentence:
Please note that while compression may save bandwidth and disk space, it incurs the cost of decompression.

Reply:
done